### PR TITLE
feat(nns): Store minted node provider rewards in stable storage

### PR DESCRIPTION
**Pull request by max-dfinity**
## Motivation

Allow historical node provider rewards to be retrieved and audited.

## What

- Create a new stable storage memory segment
- Start recording node provider rewards 
- Retrieve the latest node provider rewards from stable storage

## Followup 

After the first rewards mint to storage, we'll remove the stored rewards in heap_data.

### CODE REVIEW

File: rs/nns/governance/src/storage.rs [hunk 1]
```diff
@@ -26,6 +27,9 @@ const NEURON_FOLLOWING_INDEX_MEMORY_ID: MemoryId = MemoryId::new(11);
 const NEURON_KNOWN_NEURON_INDEX_MEMORY_ID: MemoryId = MemoryId::new(12);
 const NEURON_ACCOUNT_ID_INDEX_MEMORY_ID: MemoryId = MemoryId::new(13);
 
+const NP_REWARDS_LOG_INDEX_MEMORY_ID: MemoryId = MemoryId::new(14);
```
Comment (jasonz-dfinity) [inline rs/nns/governance/src/storage.rs:30]: NCR. I'm not always against acronyms, but in this specific case I don't feel NP is a significant enough concept.

File: rs/nns/governance/proto/ic_nns_governance/pb/storage.proto [hunk 2]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
```
Comment (jasonz-dfinity) [inline rs/nns/governance/proto/ic_nns_governance/pb/storage.proto:1]: As mentioned, I'm not seeing clearly the difference between this and the internal proto file.

**Review by jasonz-dfinity — APPROVED** [review]

File: rs/nns/governance/proto/ic_nns_governance/pb/storage.proto (hunk 2 above)
Comment (max-dfinity) [reply rs/nns/governance/proto/ic_nns_governance/pb/storage.proto:1]: Discussed offline - this is to make it simpler to segment internal storage types.  updating it to be pub(crate)

File: rs/nns/governance/src/pb/mod.rs [hunk 3]
```diff
@@ -1,6 +1,26 @@
...
+
+impl Storable for ArchivedMonthlyNodeProviderRewards {
+    fn to_bytes(&self) -> Cow<'_, [u8]> {
+        Cow::from(self.encode_to_vec())
+    }
+
+    fn from_bytes(bytes: Cow<'_, [u8]>) -> Self {
+        Self::decode(&bytes[..])
+            // Convert from Result to Self. (Unfortunately, it seems that
+            // panic is unavoid able in the case of Err.)
```
Comment (jasonz-dfinity) [inline rs/nns/governance/src/pb/mod.rs:21]: unavoidable?

**Review by jasonz-dfinity — APPROVED** [review]

File: rs/nns/governance/src/pb/mod.rs (hunk 3 above)
Comment (max-dfinity) [reply rs/nns/governance/src/pb/mod.rs]: thanks, fixed

**Pull request merged**